csrc/codegen.cpp

```diff
@@ -1001,6 +1051,50 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
   }
 
   void handle(const TernaryOp* top) final {
+    // Get vectorization information
+    if (top->out()->isA<kir::TensorIndex>()) {
+      auto out_tv = top->out()->as<kir::TensorIndex>()->view();
+      int64_t vector_word_size = ir_utils::getVectorizeSize(out_tv);
+      bool is_vector_op = vectorize_scope_ && vector_word_size != 1;
+
+      if (is_vector_op) {
+        NVF_CHECK(!top->in2()->isScalar(), "input2 should be a tensor");
+        NVF_CHECK(top->in3()->isScalar(), "input3 should be a scalar");
+        NVF_CHECK(
+            !top->out()->isScalar(),
+            "scalar output in vectorization isn't supported");
+        NVF_CHECK(
+            top->getTernaryOpType() == TernaryOpType::Where,
+            "vectorization only works on TernaryOp::where");
+
+        indent() << gen(top->in1()) << "\n";
+        indent() << kTab << "? ";
+
+        // TODO: should we have the option to specify cache level?
+        generateVectorizedLdSt(
+            top->in2(), top->out(), CacheOp::AllLevels, vector_word_size);
+        code_ << "\n";
+
+        if (out_tv->getMemoryType() == MemoryType::Local &&
+            !out_tv->isCircularBuffered()) {
+          // Vectorized initialization, explicit type conversion is needed for
+          // complex numbers
+          indent() << kTab << ": " << genVariableName(out_tv) << ".set("
```

Comment:
I'm a little confused here. Isn't this `.set` the one defined here?

https://github.com/NVIDIA/Fuser/blob/main/runtime/array.cu#L13-L18

If so, it doesn't return anything, so I thought it shouldn't compile, does it?

Reply:
> it doesn't return anything

no it doesn't. But we don't need to return anything, since we are setting outputs directly. We are also doing the same thing in the second operand when predicate == true.

```
predicate
  ? loadGlobalToLocal<...>(&dst[0], &src[i_src])
  : dst.set(0.0f) 
```
^^^ So dst is used as the target, we don't need to return anything from the ternary op.